expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlUtils.kt

```diff
@@ -0,0 +1,107 @@
+package com.expediagroup.sdk.rest.util
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import java.net.URLDecoder
+import java.nio.charset.StandardCharsets
+
+/**
+ * Parses URL path template parameters and returns a map of template segments to path segments.
+ *
+ * @param path The actual URL path.
+ * @param template The URL template with placeholders.
+ * @return A map where the keys are template segments and the values are the corresponding path segments.
+ * @throws IllegalArgumentException if either path or template is null.
+ */
+fun parseUrlPathTemplateParams(
```

Comment:
If we move these functions to `UrlUtil` or `UrlHelper` class, we can simplify the name to
```suggestion
fun parsePathParams(
```